### DISCUSSION TITLE
More details on Wan access

### DISCUSSION THREAD

**Discussion by RageXca** [haugene/docker-transmission-openvpn#2233] [Q&A] [answered]
My install works fine the vpn is also been pass on to other dockers like prowlarr/radarr/sonarr/lidarr/readarr/ombi/librespeed
Ubuntu 20.04 LTS headless server used Portainer to install a stack. I can acess them all when on LAN at home.

Issue is i wanna use my chrome addons to add torrents from wan when im at work and also access the multiple control panels ( after setting a password login for each of them )
From what i understand installing another docker with NGINX (without using     network_mode: "service:transmission-openvpn" ) and reverse proxy would do the trick.

How i would do this? 
Example of what i want on WAN i connect to port 80 (webserver) when i add /request/ to the url it connect to the ombi 192.168.1.8:3579 ( or the docker Lan ip if ran in a docker? )
then repeat those steps for each service i need including transmission remote webgui?

Guide example of config and some explanation to help me out? Most config on the net ive seen imply a domainname but from wan bt.mywanip wouldnt work since its not my own lan DNS server where i add custom domainname.
If docker a little step by step im a bit of a novice setting the stack made me learn quite a few things about docker *own lan + lan access*

This is the docker-compose ive used on Portainer to put everything in a big stack if you have any suggestions 

Compose file:
version: '2.1'
services:
 transmission-openvpn:
    volumes:
        - /files/2/finished:/data
        - /etc/localtime:/etc/localtime:ro
    environment:
        - PUID=1000
        - PGID=1000
        - CREATE_TUN_DEVICE=true
        - OPENVPN_PROVIDER=NORDVPN
        # - OPENVPN_CONFIG=default
        - NORDVPN_COUNTRY=US
        - NORDVPN_CATEGORY=legacy_p2p
        - NORDVPN_PROTOCOL=udp
        - OPENVPN_USERNAME=RETRACTED
        - OPENVPN_PASSWORD=RETRACTED
        - OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
        - WEBPROXY_ENABLED=false
        - LOCAL_NETWORK=192.168.0.0/16
        - TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false
        - DNS=8.8.8.8,8.8.4.4
    cap_add:
        - NET_ADMIN
    logging:
        driver: json-file
        options:
            max-size: 10m
    ports:
        - 9091:9091
        - 9696:9696
        - 7878:7878
        - 8989:8989
        - 8686:8686
        - 8787:8787
        - 83:80
        - 3579:3579
    restart: always
    image: haugene/transmission-openvpn
 
 prowlarr:
    image: lscr.io/linuxserver/prowlarr:develop
    network_mode: "service:transmission-openvpn"
    container_name: prowlarr
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/New_York
    volumes:
      - /home/ragex/appdata/prowlarr:/config
      - /files/2/finished:/data
    restart: unless-stopped
 
 radarr:
    image: linuxserver/radarr
    network_mode: "service:transmission-openvpn"
    container_name: radarr
    environment:
        - PUID=1000
        - PGID=1000
        - TZ=America/New_York
        - UMASK_SET=022 #optional
    volumes:
        - /home/ragex/appdata/radarr:/config
        - /files/1/Films:/movies
        - /files/2/finished:/data
    restart: unless-stopped
 
 sonarr:
    image: linuxserver/sonarr
    network_mode: "service:transmission-openvpn"
    container_name: sonarr
    environment:
        - PUID=1000
        - PGID=1000
        - TZ=America/New_York
        - UMASK_SET=022 #optional
    volumes:
        - /home/ragex/appdata/sonarr:/config
        - /files/4TB01/Animes2:/Animes01
        - /files/4TB02:/Animes02
        - /files/4TB03/series:/series02
        - /files/3:/series01
        - /files/2/finished:/data
    restart: unless-stopped
 
 lidarr:
    image: linuxserver/lidarr
    network_mode: "service:transmission-openvpn"
    container_name: lidarr
    environment:
        - PUID=1000
        - PGID=1000
        - TZ=America/New_York
        - UMASK_SET=022 #optional
    volumes:
        - /home/ragex/appdata/lidarr:/config
        - /files/1/MP3:/media
        - /files/2/finished:/data
    restart: unless-stopped
 
 readarr:
    image: lscr.io/linuxserver/readarr:develop
    network_mode: "service:transmission-openvpn"
    container_name: readarr
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/New_York
    volumes:
      - /home/ragex/appdata/readarr:/config
      - /files/1/books:/books
      - /files/2/finished:/data #optional
    restart: unless-stopped

 ombi:
    image: lscr.io/linuxserver/ombi
    container_name: ombi
    network_mode: "service:transmission-openvpn"
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/New_York
      #- BASE_URL=/ombi #optional
    volumes:
      - /home/ragex/appdata/ombi:/config
    restart: unless-stopped

 librespeed:
    image: linuxserver/librespeed
    container_name: librespeed
    network_mode: "service:transmission-openvpn"
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/New_York
      #- PASSWORD=PASSWORD
      #- DB_TYPE=sqlite #optional
      #- DB_NAME=DB_NAME #optional
      #- DB_HOSTNAME=DB_HOSTNAME #optional
      #- DB_USERNAME=DB_USERNAME #optional
      #- DB_PASSWORD=DB_PASSWORD #optional
    volumes:
      - /home/ragex/appdata/librespeed:/config
    restart: unless-stopped

**Comment by pkishino** [comment, marked as answer]
you can look at this:
https://github.com/haugene/docker-transmission-openvpn/tree/master/proxy
I agree it could do with some more documentation though
https://haugene.github.io/docker-transmission-openvpn/vpn-networking/#how_to_fix_this

**Reply by RageXca** [reply]
i got it to work with proxy and path using your "nginx config" example.
Then hours later learned about a docker called proxy manager nginx which allowed me to forward bt/lidarr and so on without editing any of them ( path for reverse proxys and such ) using domain name and its so awesome
